src/v/kafka/client/cluster.cc

```diff
@@ -137,45 +144,36 @@ ss::future<> cluster::initialize_metadata_with_seed() {
         // explicitly connect to the broker to initialize connection before
         // requesting metadata
 
-        auto request_version_f = co_await ss::coroutine::as_future(
-          get_metadata_request_version(broker, _as));
-        if (request_version_f.failed()) {
-            auto ex = request_version_f.get_exception();
+        auto describe_cluster_f = co_await ss::coroutine::as_future(
+          dispatch_describe_cluster_request(
+            broker, include_cluster_authz_t::yes));
+
+        if (describe_cluster_f.failed()) {
+            auto ex = describe_cluster_f.get_exception();
```

Comment:
With this change we no longer support clusters with earlier versions of Redpanda/Kafka. Since client implementation should remain version independent this logic is incorrect.